rules/react.js

```diff
@@ -12,25 +12,16 @@ module.exports = {
         'react/jsx-no-undef': ['error', {
             allowGlobals: true
         }],
-        'react/prefer-es6-class': 'warn',
-        'react/prefer-stateless-function': 'warn',
-        'react/prop-types': 'warn',
-
-        // findDOMNode() will be an error, use refs instead
-        'react/no-find-dom-node': 'warn',
-
-        // This will only be a warning, to promote us to write better propTypes
-        'react/forbid-prop-types': 'warn',
-
-        // This is also a warning as its a best practice to not use strings, but its not being deprecated either
-        'react/no-string-refs': 'warn',
+        'react/prefer-es6-class': 'error',
+        'react/prefer-stateless-function': 'error',
+        'react/prop-types': 'error',
+        'react/no-find-dom-node': 'error',
+        'react/forbid-prop-types': 'error',
+        'react/no-string-refs': 'error',
 
         // Rather then blindly enforcing destructuring assignments, we'll trust the author's best judgement on when
         // to make use of them, and when not; see https://github.com/Expensify/Style-Guide/pull/60 for more details
-        'react/destructuring-assignment': 'off',
-
-        // We want to force the use of stateless functions when we can
-        'react/prefer-stateless-function': 'error',
+        'react/destructuring-assignment': 'warn',
```

Comment:
I'm changing this from `off` to `warn` to make it more visible and to gently push back against doing this.

Reply:
I thought this lint rule enforced destructuring assignments always (which we found to be annoying and at times bad for code quality). By making it a warning won't we be encouraging people (who choose to address not only errors but warning as well) to use destructuring at will?

Reply:
Oh, good call! By default, it will enforce that you always destructure (which we don't want), but we can flip that default to be `never` and I think having that as a warning would be great. Would you be OK with that?

Reply:
When would this warn? Just want to make sure I understand the change here.

Reply:
Oh ok I was confused about what this rule did and thought it was warning when we DID NOT destructure all the things. I actually liked how we had it before where it was author's choice and don't see any reason to push back on people destructuring things. 

My main issue with the AirBnB rule is that it forced us to use destructuring in cases where it made no sense to do so. I think sometimes it makes sense to use destructuring and other times not so much so our rule should reflect that and not take the complete opposite view on destructuring. 

We should also first change rules in the style guide before having eslint enforce them... our current stance on destructuring is here..

https://github.com/Expensify/Style-Guide/blob/master/javascript.md#destructuring

This eslint rule now introduces confusion about what the right thing to do is and I think our existing guideline is clear about when it's useful and when it decreases code clarity.

Reply:
Yeah, those are good points. This would be another one of those things where having our own eslint plugin could help us better enforce our own styles.

So, my takeaway here is that... even though it's a warning, we still prefer to NOT TO favor one method over the other? If that's the case, I can turn this back to `off`.

cc @iwiznia @nkuoch @deetergp @bondydaa @stitesExpensify @pecanoro (not sure who else to ping, but I want to get multiple opinions, feel free to ping others)

Reply:
I think I'm in agreement 🤔 that is if I'm understanding correctly - the plan is:

1. change this rule back to `off`
1. Leave https://github.com/Expensify/Style-Guide/blob/master/javascript.md#destructuring as is and allow desctructuring of props in stateless components

If that is not the plan then I guess I disagree and would rather do the above 😅

Reply:
The alternative is: make ESLint throw a warning if you don't do it the "preferred" way (which I guess we don't have a good way of determining what's good or bad without coding our own ESLint plugin). Then you (the code author) can still choose whether or not you make the ESLint warning go away.

Reply:
would prefer if the warnings we do have are "nice to haves" and not unclear whether they are a problem or not. super long lines of code are a problem but not a big problem... not sure I'd say the same about this.

Reply:
OK, I'm not sure what you are all suggesting then. If you'd like to make change to this rule, go ahead and open up a PR to change it 👍